challenge.rb

```diff
@@ -1,26 +1,26 @@
 
 def capitalize_each_string(input)
-  #implement your solution here
+  input.map{|x| x.capitalize}
 end
 
 def fetch_the_dog(input)
-  #implement your solution here
+  input.select{|x| x == "dog"}
 end
 
 def no_dogs_allowed(input)
-  #implement your solution here
+  input.reject{|x| x == "dog"}
 end
 
 def count_the_animals(input)
-  #implement your solution here
+  input.count
 end
 
 def fetch_the_first_two(input)
-  #implement your solution here
+  input.take(2)
 end
 
 def fetch_CD_animals(input)
-  #implement your solution here
+  input.find_all{|i| i == "cat" or i == "dog"}
```

Comment:
This will work, but note the method name is "fetch_CD_animals". Were I to encounter this method in a development gig, I'd assume its intent is to return all animals who's names start with the letter C or D. Then I'd run your implementation and wonder why I wasn't getting back any ducks, dingos, cows, crows, and all sorts of other animals. :stuck_out_tongue_closed_eyes: 

So, how could you rewrite this method to more closely achieve what its name implies?